bentoml/_internal/bento/build_config.py

```diff
@@ -525,11 +549,20 @@ def write_to_bento(self, bento_fs: FS, build_ctx: str) -> None:
         echo "WARNING: using BentoML version ${existing_bentoml_version}"
     fi
 else
-pip install bentoml=="$BENTOML_VERSION"
+    pip install bentoml=="$BENTOML_VERSION"
 fi
 """
             )
-            f.write(install_script_content)
+            if self.components:
+                components = map(
+                    lambda t: t[0], filter(lambda t: t[1], self.components.items())
+                )
```

Comment:
List comprehensions are the only feature of Python that I really like ;)

```suggestion
                components = [component for component, enabled in self.components.items() if enabled]
```

Reply:
fair.